source/common/upstream/ring_hash_lb.cc

```diff
@@ -122,37 +124,31 @@ RingHashLoadBalancer::Ring::Ring(const NormalizedHostWeightVector& normalized_ho
   // For stats reporting, keep track of the minimum and maximum actual number of hashes per host.
   // Users should hopefully pay attention to these numbers and alert if min_hashes_per_host is too
   // low, since that implies an inaccurate request distribution.
-  char hash_key_buffer[196];
+
+  absl::InlinedVector<char, 196> hash_key_buffer;
   double current_hashes = 0.0;
   double target_hashes = 0.0;
   uint64_t min_hashes_per_host = ring_size;
   uint64_t max_hashes_per_host = 0;
   for (const auto& entry : normalized_host_weights) {
     const auto& host = entry.first;
-    const std::string& address_string = host->address()->asString();
-    uint64_t offset_start = address_string.size();
-
-    // Currently, we support both IP and UDS addresses. The UDS max path length is ~108 on all Unix
-    // platforms that I know of. Given that, we can use a 196 char buffer which is plenty of room
-    // for UDS, '_', and up to 21 characters for the node ID. To be on the super safe side, there
-    // is a RELEASE_ASSERT here that checks this, in case someone in the future adds some type of
-    // new address that is larger, or runs on a platform where UDS is larger. I don't think it's
-    // worth the defensive coding to deal with the heap allocation case (e.g. via
-    // absl::InlinedVector) at the current time.
-    RELEASE_ASSERT(
-        address_string.size() + 1 + StringUtil::MIN_ITOA_OUT_LEN <= sizeof(hash_key_buffer), "");
-    memcpy(hash_key_buffer, address_string.c_str(), offset_start);
-    hash_key_buffer[offset_start++] = '_';
+    const std::string& address_string =
+        use_hostname ? host->hostname() : host->address()->asString();
+
+    hash_key_buffer.assign(address_string.begin(), address_string.end());
+    hash_key_buffer.emplace_back('_');
+    auto offset_start = hash_key_buffer.end();
 
     // As noted above: maintain current_hashes and target_hashes as running sums across the entire
     // host set. `i` is needed only to construct the hash key, and tally min/max hashes per host.
     target_hashes += scale * entry.second;
     uint64_t i = 0;
     while (current_hashes < target_hashes) {
-      const uint64_t total_hash_key_len =
-          offset_start +
-          StringUtil::itoa(hash_key_buffer + offset_start, StringUtil::MIN_ITOA_OUT_LEN, i);
-      absl::string_view hash_key(hash_key_buffer, total_hash_key_len);
+      std::string i_str = absl::StrCat("", i);
```

Comment:
You can also use `absl::SimpleAtoi` here. Also, while you are here please constify all variables that you can.